lib/irq.c

```diff
@@ -189,7 +189,12 @@ irqs_set_data_none(vfu_ctx_t *vfu_ctx, struct vfio_irq_set *irq_set)
     eventfd_t val;
 
     for (i = irq_set->start; i < (irq_set->start + irq_set->count); i++) {
-        efd = vfu_ctx->irqs->efds[i];
+        if (irq_set->index == VFIO_PCI_ERR_IRQ_INDEX)
```

Comment:
please make this be a common helper - and use switch() ?

Reply:
Ok, I will replace all if else with switch().

Reply:
I guess for `ERR` and `REQ` case, if wouldn't make sense for `irq_set->count` not to be 1? I suppose it might be technically correct, was wondering whether we could simplify the code because of this.